modules/compute/src/main/java/org/apache/ignite/internal/compute/task/TaskExecutionInternal.java

```diff
@@ -81,6 +84,8 @@ public class TaskExecutionInternal<I, M, T, R> implements TaskExecution<R> {
 
     private final AtomicBoolean isCancelled;
 
+    private final AtomicReference<Marshaller<R, byte[]>> reduceResultMarshallerRef = new AtomicReference<>();
```

Comment:
We only use `get` and `set` from the `AtomicReference`, so it can be replaced with a `volatile` field.